recipes/log4cxx/all/conanfile.py

```diff
@@ -0,0 +1,146 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+import textwrap
+
+required_conan_version = ">=1.33.0"
+
+
+class Log4cxxConan(ConanFile):
+    name = "log4cxx"
+    description = "Logging framework for C++ patterned after Apache log4j"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "Apache-2.0"
+    homepage = "https://logging.apache.org/log4cxx"
+    topics = ("logging", "log")
+    exports_sources = ("CMakeLists.txt", "patches/**")
+    generators = "cmake", "cmake_find_package", "pkg_config"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires("apr/1.7.0")
+        self.requires("apr-util/1.6.1")
+        self.requires("expat/2.4.1")
+        if self.settings.os != "Windows":
+            self.requires("odbc/2.3.9")
+
+    def build_requirements(self):
+        if self.settings.os != "Windows":
+            self.build_requires("pkgconf/1.7.4")
```

Comment:
pkgconf doesn't seem to be used by upstream CMakeLists

Reply:
No, you're right, missed that: https://github.com/apache/logging-log4cxx/blob/70199c597ee02f5582dcfee47b8947d7c4abaa33/src/main/include/CMakeLists.txt#L74-L79